src/kernel.cr

```diff
@@ -178,6 +178,14 @@ def abort(message, status = 1) : NoReturn
   exit status
 end
 
+# Expresses an unreachable code explicitly.
+#
+# When it is called, it raises `UnreachableError` with given *message*.
+# However you will never see it if you use this method accurately.
+def unreachable!(message = "unreachable") : NoReturn
+  raise UnreachableError.new("BUG: #{message}")
```

Comment:
I wouldn't hardcode "BUG: ..." in every unreachable message. Please move "Bug: " inside the default value.

Reply:
done.